e2e-tests/src/test/scala/pl/touk/nussknacker/BatchDataGenerationSpec.scala

```diff
@@ -0,0 +1,77 @@
+package pl.touk.nussknacker
+
+import io.circe.syntax.EncoderOps
+import io.restassured.RestAssured.`given`
+import io.restassured.module.scala.RestAssuredSupport.AddThenToResponse
+import org.scalatest.freespec.AnyFreeSpecLike
+import org.scalatest.matchers.should.Matchers
+import pl.touk.nussknacker.config.WithE2EInstallationExampleRestAssuredUsersExtensions
+import pl.touk.nussknacker.engine.build.ScenarioBuilder
+import pl.touk.nussknacker.engine.spel.SpelExtension._
+import pl.touk.nussknacker.test.{NuRestAssureExtensions, NuRestAssureMatchers, VeryPatientScalaFutures}
+import pl.touk.nussknacker.ui.process.marshall.CanonicalProcessConverter.toScenarioGraph
+
+class BatchDataGenerationSpec
+    extends AnyFreeSpecLike
+    with DockerBasedInstallationExampleNuEnvironment
+    with Matchers
+    with VeryPatientScalaFutures
+    with NuRestAssureExtensions
+    with NuRestAssureMatchers
+    with WithE2EInstallationExampleRestAssuredUsersExtensions {
+
+  private val simpleBatchTableScenario = ScenarioBuilder
+    .streaming("SumTransactions")
+    .source("sourceId", "table", "Table" -> "'transactions'".spel)
+    .emptySink("end", "dead-end")
+
+  private val designerServiceUrl = "http://localhost:8080"
+
+  "Generate file endpoint should generate records with randomized values for scenario with table source" in {
+    val expectedRecordRegex =
```

Comment:
now, we can inline the "json-regex" ;) The reader will pay attention to this when they are reading the "then" part. Currently, the reader sees it, when they start to read code of the test, but it'll be relevant later - almost at the end of the test.

Reply:
right, done :+1: